go-controller/pkg/metrics/ipsec.go

```diff
@@ -0,0 +1,33 @@
+package metrics
+
+import (
+	"fmt"
+	"regexp"
+	"strings"
+
+	"k8s.io/apimachinery/pkg/util/sets"
+)
+
+type ipsecClient func(args ...string) (string, string, error)
+
+func listEstablishedIPsecTunnels(ipsec ipsecClient) (sets.Set[string], error) {
+	ipsecTunnels := sets.Set[string]{}
+	stdout, stderr, err := ipsec("showstates")
+	if err != nil {
+		return ipsecTunnels, fmt.Errorf("failed to retrieve ipsec states %v: %v", stderr, err)
+	}
+	if stdout == "" {
+		return ipsecTunnels, nil
+	}
+
+	lines := strings.Split(string(stdout), "\n")
+	for _, line := range lines {
+		if strings.Contains(line, "ESTABLISHED_CHILD_SA") {
+			matches := regexp.MustCompile(`"([^"]*)"`).FindAllStringSubmatch(line, -1)
```

Comment:
maybe write a comment on the expression?

Reply:
done.

Reply:
hmm - matching on things just having quotes doesnt sound great. I know its unlikely the output changes but it seems super weak to match on all quoted words. I thought we'd only expect one at least but i see you dont limit the number of matches. I am hoping theres json output for this cli.

Reply:
there is no json output option for ipsec commands for now.